ystia/yorc
# Technical/gh 297 update metrics

# Pull Request description

## Description of the change

### What I did
Upgrade armon/go-metrics to v0.30.
Redefine Yorc metrics names. Also use labels.
Update Yorc Telemetry chapter in Yorc documentation.

### How I did it
In case of metrics that are identified by a name + some labels, I replaced the previousely called go-metrics function calls by the their variant that supports labels.
For example :
metrics.IncrCounter([]string{"name-part1", "name-part2", "status"}), 1)
replaces by
metrics.IncrCounterWithLabels(([]string{"name-part1", "name-part2"}), 1, lablels)
where 
labels contain a label with Name = "status" and Value = getStatus()

### How to verify it
Yorc telemetry config:
`"telemetry": {
    "service_name": "myyorc",
    "disable_hostname": true,
    "disable_go_runtime_metrics": true,
    "expose_prometheus_endpoint": true
  }`

Deploy an application.

Check for metrcis on Yorc endpoint:

`$ curl --noproxy '*' -v -H "Content-Type: application/json" "http://localhost:8800/metrics`
Example of metrics:
myyorc_http_duration_sum{method="GET",path="metrics"} 222.54512375593185
myyorc_http_duration_count{method="GET",path="metrics"} 68
myyorc_http_total{method="GET",path="deployments",status="0"} 1

Install and start Prometheus with scrape config:
scrape_configs section in prometheus.yml config file:
  - job_name: 'yorc'
    static_configs:
    - targets: ['localhost:8800']

Check for metrcis on endpoint exposed for Prometheus:

`$ curl --noproxy '*' -v -H "Content-Type: application/json" "http://localhost:9090/metrics`

or execute a query in Prometheus console, for example:
`myyorc_http_total`

### Description for the changelog

## Applicable Issues
https://github.com/ystia/yorc/issues/297

https://github.com/ystia/yorc/issues/507
